src/Hit.cc

```diff
@@ -40,167 +52,170 @@ Hit::~Hit() {}
+    m_passiveHitType      = h.m_passiveHitType;
+    m_hitPassiveElem      = h.m_hitPassiveElem;
+    m_isPixel             = h.m_isPixel;
+    m_resolutionRPhi      = h.m_resolutionRPhi;
+    m_resolutionZ         = h.m_resolutionZ;
+    m_isPixelIntersticeVol= false;
+    m_isPixelTrackingVol  = false;
+    m_isIntersticeVol     = false;
+    m_isOuterTrackingVol  = false;
+    m_isTotalTrackingVol  = false;
+}
+
+/**
+ * Constructor for a hit on an inactive surface at a given [rPos, zPos] from the origin
+ */
+Hit::Hit(double rPos, double zPos, const insur::InactiveElement* myPassiveElem, HitPassiveType passiveHitType) {
+    m_detName             = "Undefined";
+    m_distance            = sqrt(rPos*rPos + zPos*zPos);
+    m_rPos                = rPos;
+    m_zPos                = zPos;
+    m_activity            = HitActivity::Inactive;
+    m_activeHitType       = HitType::NONE;
+    m_hitModule           = nullptr;
+    m_track               = nullptr;
+    m_isTrigger           = false;
+    m_passiveHitType      = passiveHitType;
+    setHitPassiveElement(myPassiveElem);
+    m_isPixel             = false;
+    m_resolutionRPhi      = 0;
+    m_resolutionZ         = 0;
+    m_isPixelIntersticeVol= false;
+    m_isPixelTrackingVol  = false;
+    m_isIntersticeVol     = false;
+    m_isOuterTrackingVol  = false;
+    m_isTotalTrackingVol  = false;
+
+    if (m_passiveHitType==HitPassiveType::BeamPipe) m_detName = "BeamPipe";
+    if (m_passiveHitType==HitPassiveType::IP)       m_detName = "IP";
+    if (m_passiveHitType==HitPassiveType::Support)  m_detName = "Support";
+    if (m_passiveHitType==HitPassiveType::Service)  m_detName = "Service";
+}
+
+/**
+ * Constructor for a hit on a given module at [rPos, zPos] (cylindrical position) from the origin
  * @param myModule pointer to the module with the hit 
  */
-Hit::Hit(double myDistance, Module* myModule, HitType activeHitType) {
-    distance_ = myDistance;
-    objectKind_ = Active;
-    objectCategory_ = Act;
-    hitInactiveElement_ = NULL;
-    orientation_ = Undefined; 
-    isTrigger_ = false;
-    isPixel_ = false;
-    isPixelIntersticeVolume_ = false;
-    isPixelTrackingVolume_ = false;
-    isIntersticeVolume_ = false;
-    isOuterTrackingVolume_ = false;
-    isTotalTrackingVolume_ = false;
-    isIP_ = false;
+Hit::Hit(double rPos, double zPos, const DetectorModule* myModule, HitType activeHitType) {
+    m_detName          = "Undefined";
+    m_distance         = sqrt(rPos*rPos + zPos*zPos);
+    m_rPos             = rPos;
+    m_zPos             = zPos;
+    m_activity         = HitActivity::Active;
+    m_activeHitType    = activeHitType;
     setHitModule(myModule);
-    myTrack_ = NULL;
-    activeHitType_ = activeHitType;
+    m_track            = nullptr;
+    m_isTrigger        = false;
+    m_passiveHitType   = HitPassiveType::Undefined;
+    m_hitPassiveElem   = nullptr;
+    m_isPixel          = false;
+    m_resolutionRPhi   = 0;
+    m_resolutionZ      = 0;
+
```

Comment:
ERROR !
Should add here : 
m_isPixelIntersticeVol= false;
+    m_isPixelTrackingVol  = false;
+    m_isIntersticeVol     = false;
+    m_isOuterTrackingVol  = false;
+    m_isTotalTrackingVol  = false;